dotCMS/src/main/java/com/dotmarketing/portlets/structure/action/EditFieldAction.java

```diff
@@ -318,6 +319,43 @@ private boolean _saveField(ActionForm form, ActionRequest req, ActionResponse re
                  }
             }
 
+            // Validate values entered for decimal/number type check box field
+            if (field.getFieldType().equals(FieldType.RADIO.toString())){
+                String values = fieldForm.getValues();
+                String temp = values.replaceAll("\r\n","|");
+                String[] tempVals = StringUtil.split(temp.trim(), "|");
+
+                if (tempVals.length % 2 == 1){
+                    SessionMessages.add(req, "error", "message.structure.missingdatavalue");
+                    return false;
+                }
+
+                try {
+                    if(dataType.equals(Field.DataType.FLOAT.toString())){
+                        for(int i=1;i<tempVals.length;i+= 2){
+                            Float.parseFloat(tempVals[i]);
+                        }
+                    }else if(dataType.equals(Field.DataType.INTEGER.toString())){
+                        for(int i=1;i<tempVals.length;i+= 2){
+                            Integer.parseInt(tempVals[i]);
+                        }
+                    }else if(dataType.equals(Field.DataType.BOOL.toString())){
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this use of "Field"; it is deprecated. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this use of "DataType"; it is deprecated. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this use of "BOOL"; it is deprecated. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)